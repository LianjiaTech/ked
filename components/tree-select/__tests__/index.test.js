import React from 'react';
import { mount } from 'enzyme';
import TreeSelect, { TreeNode } from '..';
import mountTest from '../../../tests/shared/mountTest';

describe('TreeSelect', () => {
  mountTest(TreeSelect);

  describe('showSearch', () => {
    it('keep default logic', () => {
      const single = mount(<TreeSelect open />);
      expect(single.find('.ant-select-search__field').length).toBeFalsy();

      const multiple = mount(<TreeSelect multiple open />);
      expect(multiple.find('.ant-select-search__field').length).toBeTruthy();
    });
  });

  describe('TreeSelect Custom Icons', () => {
    it('should support customized icons', () => {
      const wrapper = mount(
        <TreeSelect
          showSearch
          clearIcon={<span>clear</span>}
          removeIcon={<span>remove</span>}
          value={['leaf1', 'leaf2']}
          placeholder="Please select"
          multiple
          allowClear
          treeDefaultExpandAll
        >
          <TreeNode value="parent 1" title="parent 1" key="0-1">
            <TreeNode value="parent 1-0" title="parent 1-0" key="0-1-1">
              <TreeNode value="leaf1" title="my leaf" key="random" />
              <TreeNode value="leaf2" title="your leaf" key="random1" />
            </TreeNode>
          </TreeNode>
        </TreeSelect>,
      );

      expect(wrapper.render()).toMatchSnapshot();
    });
  });
});
