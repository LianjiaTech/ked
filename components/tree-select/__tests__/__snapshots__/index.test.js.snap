// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeSelect TreeSelect Custom Icons should support customized icons 1`] = `
<span
  aria-haspopup="listbox"
  class="ant-select ant-select-enabled ant-select-allow-clear"
  role="combobox"
  tabindex="-1"
>
  <span
    class="ant-select-selection ant-select-selection--multiple"
  >
    <ul
      class="ant-select-selection__rendered"
      role="menubar"
    >
      <li
        class="ant-select-selection__choice"
        role="menuitem"
        style="user-select: none;"
        unselectable="unselectable"
      >
        <span
          class="ant-select-selection__choice__remove"
        >
          <span>
            remove
          </span>
        </span>
        <span
          class="ant-select-selection__choice__content"
        >
          leaf1
        </span>
      </li>
      <li
        class="ant-select-selection__choice"
        role="menuitem"
        style="user-select: none;"
        unselectable="unselectable"
      >
        <span
          class="ant-select-selection__choice__remove"
        >
          <span>
            remove
          </span>
        </span>
        <span
          class="ant-select-selection__choice__content"
        >
          leaf2
        </span>
      </li>
      <li
        class="ant-select-search ant-select-search--inline"
      >
        <span
          class="ant-select-search__field__wrap"
        >
          <input
            aria-autocomplete="list"
            aria-label="filter select"
            aria-multiline="false"
            class="ant-select-search__field"
            style="width: 0px;"
            type="text"
            value=""
          />
          <span
            class="ant-select-search__field__mirror"
          >
            Â 
          </span>
        </span>
      </li>
    </ul>
    <span
      class="ant-select-selection__clear"
    >
      <span>
        clear
      </span>
    </span>
    <span
      class="ant-select-search__field__placeholder"
      style="display: none;"
    >
      Please select
    </span>
  </span>
</span>
`;
